api/meetings/views.py

```diff
@@ -0,0 +1,238 @@
+        queryset = self.annotate_queryset_with_download_count(queryset)
+        return queryset
+
+    def annotate_queryset_with_meeting_category(self, meeting, queryset):
+        """
+        Annotates queryset with meeting_category - if submission1 tag exists, use this,
+        otherwise assume default submission2 tag
+        """
+        # Setup meeting category subquery (really existence of certain tags)
+        category_1 = meeting.field_names.get('submission1', 'poster')
+        category_2 = meeting.field_names.get('submission2', 'talk')
+        tag_subquery = Tag.objects.filter(
+            abstractnode_tagged=OuterRef('pk'),
+            name=category_1,
+        ).values_list('name', flat=True)
+
+        queryset = queryset.annotate(cat_one_count=Count(Subquery(tag_subquery))).annotate(
+            meeting_category=Case(
+                When(cat_one_count=1, then=Value(category_1)),
+                default=Value(category_2),
+                output_field=CharField(),
+            ),
+        )
+        return queryset
+
+    def annotate_queryset_with_author_name(self, queryset):
+        """
+        Annotates queryset with author_name_category - it is the family_name if it exists, otherwise,
+        the fullname is used
+        """
+        # Setup author name subquery (really first bibliographic contributor)
+        contributors = Contributor.objects.filter(
+            visible=True,
+            node_id=OuterRef('pk'),
+        ).order_by('_order')
+
+        queryset = queryset.annotate(
+            author_family_name=Subquery(contributors.values(('user__family_name'))[:1]),
+        ).annotate(
+            author_full_name=Subquery(contributors.values(('user__fullname'))[:1]),
+        ).annotate(
+            author_name=Case(
+                When(author_family_name='', then=F('author_full_name')),
+                default=F('author_family_name'),
+                output_field=CharField(),
+            ),
+        )
+        return queryset
+
+    def annotate_queryset_with_download_count(self, queryset):
+        """
+        Annotates queryset with download count of first osfstorage file
+        """
+        pages = PageCounter.objects.annotate(
+            node_id=Substr('_id', 10, 5),
+            file_id=Substr('_id', 16),
+            _id_length=Length('_id'),
+        ).filter(
+            _id__icontains='download',
+            node_id=OuterRef('guids___id'),
+            file_id=OuterRef('file_id'),
+        ).exclude(_id_length__gt=39)
+
+        file_subqs = OsfStorageFile.objects.filter(
+            target_content_type_id=ContentType.objects.get_for_model(AbstractNode),
+            target_object_id=OuterRef('pk'),
+        ).order_by('created')
+
+        queryset = queryset.annotate(
+            file_id=Subquery(file_subqs.values('_id')[:1]),
+        ).annotate(
+            download_count=Coalesce(Subquery(pages.values('total')[:1]), Value(0)),
+        )
+        return queryset
```

Comment:
Hopefully by the time this would break, we will be never using Pagecounter again. Could you put in a comment to say the expected failure cases in case someone does end up tracing code to near here?

Reply:
Done!

Reply:
Also, my work in https://openscience.atlassian.net/browse/ENG-122 could get rid of this!